contracts/src/PDPService.sol

```diff
@@ -3,15 +3,151 @@ pragma solidity ^0.8.13;
 
 import "@openzeppelin/contracts/access/Ownable.sol";
 
-contract PDPService is Ownable {
+contract PDPService {
     // Types 
+    // TODO PERF: https://github.com/FILCAT/pdp/issues/16#issuecomment-2329836995 
+    struct Cid {
+        bytes data;
+    }
 
     // State fields
-    uint256 public challengeFinality;
+
+    /*
+    A proof set is the metadata required for tracking data for proof of possession.
+    It maintains a list of CIDs of data to be proven and metadata needed to 
+    add and remove data to the set and prove possession efficiently.
+
+    ** logical structure of the proof set**
+    /* 
+    struct ProofSet {
+        Cid[] roots; 
+        uint256[] sizes;
+        uint256[] sumTree;
+        uint256 size;
+        address owner;
+        nextRootID uint64;
+    }
+    ** PDP service contract tracks many possible proof sets **
+    []ProofSet proofsets
+
+    To implement this logical structure in the solidity data model we have
+    two arrays tracking the singleton fields and three two dimensional arrays
+    tracking the growing data of the proof set.  The first index is the proof set id
+    and the second index is the index of the data in the array.
+
+    Invariant: rootCids.length == rootSizes.length == sumTreeSizes.length
+    */
+
+    // Network epoch delay between last proof of possession and next 
+    // randomness sampling for challenge generation
+    uint256 challengeFinality;
+   
+    // TODO PERF: https://github.com/FILCAT/pdp/issues/16#issuecomment-2329838769
+    uint64 nextProofSetId;
+    mapping(uint256 => mapping(uint256 => Cid)) rootCids; 
+    mapping(uint256 => mapping(uint256 => uint256)) rootSizes;
+    mapping(uint256 => mapping(uint256 => uint256)) sumTreeSizes;
```

Comment:
Specifically for the sum tree, maybe an array in the inner collection makes sense? It'll always be dense, right?